app.rb

```diff
@@ -50,4 +58,36 @@ def debug_params
     list.save!
   end
   redirect back
+end
+
+post "lists/delete" do
+  debug_params
+end  
```

Comment:
this seems difficult right? you would have to load all lists and assign new ids to each one of them? for example, if you have 4 lists 0.md, 1.md, 2.md and 3.md; then if you delete 1.md, you will have to save the remaining 3 lists as 0.md, 1.md and 2.md. right? 

here are some HINTS:

1. you can create a DELETE button similar to DONE button. when you press Delete, it should submit the List id to delete

2. you can implement the controller code similar to the following:

```ruby
# this is just a guided example. the code is not tested

get "/lists/:id/delete" do
  list = List.new(params["delete"])
  list.delete! # implement a method to delete the file here
  # then reindex:
  List.load_all_lists.each_with_index do |list, index|
    list.id = list.index
    list.save
  end
```